### PR TITLE
Remove e2e build tag

### PR DESCRIPTION
**Pull request by enxebre**
**What this PR does / why we need it**:
I question the e2e build tags are giving us anything today and they get my IDE confused when navigating through symbols.
/hold
For others to have the chance to prove its value.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -154,7 +154,7 @@ app-sre-saas-template: hypershift
 # Run tests
 .PHONY: test
 test:
-	$(GO) test -race -count=25 -timeout=30m ./... -coverprofile cover.out
+	$(GO) test -race -count=25 -timeout=30m $(shell go list ./... | grep -v e2e) -coverprofile cover.out
```
Comment (bryan-cox) [inline Makefile:157]: nit: Not sure if we can change it, but it would be nice if the list of directories doesn't display when running `make test`.

**Review by bryan-cox — COMMENTED** [review]
This seems good to me. I checked out the PR and ran some of the commands to see if there were any differences or what not: `make build`, `make e2e`, `make test`.

File: Makefile (hunk 1 above)
Comment (jparrill) [reply Makefile:157]: You can do that just adding the "@" in front of the "$(GO)" command:

```
@$(GO) test -race -count=25 -timeout=30m $(shell go list ./... | grep -v e2e) -coverprofile cover.out
```

**Review by jparrill — COMMENTED** [review]
apart of the comment, lgtm so far

**Pull request closed without merging**